pkg/configurator/crd_client.go

```diff
@@ -190,3 +250,83 @@ func meshConfigUpdatedMessageHandler(psubMsg *events.PubSubMessage) {
 			psubMsg.AnnouncementType)
 	}
 }
+
+func (c *CRDClient) getMeshConfigCacheKey() string {
+	return fmt.Sprintf("%s/%s", c.osmNamespace, c.meshConfigName)
+}
+
+// Returns the current ConfigMap
+func (c *CRDClient) getMeshConfig() *osmConfig {
+	meshConfigCacheKey := c.getMeshConfigCacheKey()
+	item, exists, err := c.cache.GetByKey(meshConfigCacheKey)
+	if err != nil {
+		log.Error().Err(err).Msgf("Error getting MeshConfig from cache with key %s", meshConfigCacheKey)
+		return &osmConfig{}
+	}
+
+	var meshConfig *v1alpha1.MeshConfig
+	if !exists {
+		log.Warn().Msgf("MeshConfig %s does not exist. Default config values will be used.", meshConfigCacheKey)
+		meshConfig = &v1alpha1.MeshConfig{}
+	} else {
+		meshConfig = item.(*v1alpha1.MeshConfig)
+	}
+
+	return parseOSMMeshConfig(meshConfig)
+}
+
+// This struct must match the shape of the "osm-config" ConfigMap
+// which was created in the OSM namespace.
+type osmConfig struct {
```

Comment:
Like noted above, this seems unnecessary. The MeshConfig object should be directly parseable.